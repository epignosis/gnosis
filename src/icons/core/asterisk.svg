<svg id="ic_advanced_search" data-name="ic_advanced search" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path id="Path_2112" data-name="Path 2112" d="M22.468-4.255,14.25-9l8.218-4.744a.563.563,0,0,0,.206-.768l-.562-.974a.563.563,0,0,0-.768-.206l-8.218,4.744v-9.489A.563.563,0,0,0,12.562-21H11.437a.563.563,0,0,0-.562.563v9.489L2.657-15.693a.563.563,0,0,0-.768.206l-.562.974a.563.563,0,0,0,.206.768L9.75-9,1.532-4.255a.563.563,0,0,0-.206.768l.562.974a.563.563,0,0,0,.768.206l8.218-4.744V2.437A.563.563,0,0,0,11.437,3h1.125a.563.563,0,0,0,.562-.562V-7.051l8.218,4.744a.563.563,0,0,0,.768-.206l.563-.974A.563.563,0,0,0,22.468-4.255Z" transform="translate(0 21)"/>
  <rect id="Rectangle_1183" data-name="Rectangle 1183" width="24" height="24" transform="translate(0 0)" fill="none"/>
</svg>
