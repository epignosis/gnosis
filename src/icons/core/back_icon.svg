<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 25 25">
  <g id="icon_goback" transform="translate(-1097 -1370)">
    <rect id="Rectangle_1063" data-name="Rectangle 1063" width="25" height="25" transform="translate(1097 1370)" fill="#fff"/>
    <path id="Path_132" data-name="Path 132" d="M4.765-1.616l.613-.613a.375.375,0,0,0-.005-.536L2.851-5.187H13.625A.375.375,0,0,0,14-5.562v-.875a.375.375,0,0,0-.375-.375H2.851L5.373-9.235a.375.375,0,0,0,.005-.536l-.613-.613a.375.375,0,0,0-.53,0L.116-6.265a.375.375,0,0,0,0,.53L4.235-1.616A.375.375,0,0,0,4.765-1.616Z" transform="translate(1103 1388)"/>
  </g>
</svg>
