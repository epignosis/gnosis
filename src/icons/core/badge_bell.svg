<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 95 95">
  <g id="icon_dontmiss" transform="translate(-126.242)">
    <circle id="Ellipse_220" data-name="Ellipse 220" cx="47.5" cy="47.5" r="47.5" transform="translate(126.242)" fill="#0e59bb"/>
    <g id="Group_13104" data-name="Group 13104" transform="translate(145 19)">
      <path id="Path_2101" data-name="Path 2101" d="M14.341-16.46C12.653-18.274,9.493-21,9.493-29.945A13.8,13.8,0,0,0-1.689-43.507v-1.821a2.8,2.8,0,0,0-2.8-2.8,2.8,2.8,0,0,0-2.8,2.8v1.821A13.8,13.8,0,0,0-18.462-29.945c0,8.941-3.16,11.671-4.848,13.485a2.731,2.731,0,0,0-.753,1.9,2.8,2.8,0,0,0,2.806,2.8H12.288a2.8,2.8,0,0,0,2.806-2.8A2.73,2.73,0,0,0,14.341-16.46Zm-32.5.5c1.855-2.445,3.882-6.5,3.892-13.934,0-.017-.005-.033-.005-.051a9.789,9.789,0,0,1,9.789-9.789A9.789,9.789,0,0,1,5.3-29.945c0,.017-.005.033-.005.051.01,7.438,2.037,11.49,3.892,13.934ZM-4.484-3.375A5.593,5.593,0,0,0,1.107-8.969H-10.075A5.593,5.593,0,0,0-4.484-3.375Z" transform="translate(33.063 54.125)" fill="#fff"/>
      <rect id="Rectangle_1633" data-name="Rectangle 1633" width="57" height="57" transform="translate(0.242)" fill="none"/>
    </g>
  </g>
</svg>
