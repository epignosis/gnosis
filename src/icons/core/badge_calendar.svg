<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 95 95">
  <g id="icon_today" transform="translate(-1390 -977)">
    <circle id="Ellipse_221" data-name="Ellipse 221" cx="47.5" cy="47.5" r="47.5" transform="translate(1390 977)" fill="#0e59bb"/>
    <path id="Path_2102" data-name="Path 2102" d="M10.117-42.656h-4.1V-47.1A1.028,1.028,0,0,0,4.99-48.125H1.572A1.028,1.028,0,0,0,.547-47.1v4.443H-10.391V-47.1a1.028,1.028,0,0,0-1.025-1.025h-3.418A1.028,1.028,0,0,0-15.859-47.1v4.443h-4.1a4.1,4.1,0,0,0-4.1,4.1V-8.477a4.1,4.1,0,0,0,4.1,4.1H10.117a4.1,4.1,0,0,0,4.1-4.1V-38.555A4.1,4.1,0,0,0,10.117-42.656ZM9.6-8.477H-19.448a.514.514,0,0,1-.513-.513V-34.453H10.117V-8.989A.514.514,0,0,1,9.6-8.477Z" transform="translate(1442.063 1050.125)" fill="#fff"/>
    <rect id="Rectangle_1634" data-name="Rectangle 1634" width="57" height="57" transform="translate(1409 996)" fill="none"/>
  </g>
</svg>
