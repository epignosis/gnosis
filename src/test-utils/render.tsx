import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
