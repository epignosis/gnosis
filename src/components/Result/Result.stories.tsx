import React from "react";
import { Story } from "@storybook/react";
import Result, { ResultProps } from "./Result";
import { InfoSVG } from "@icons/core";

export default {
  title: "Components/Result",
  component: Result,
  argTypes: {
    size: {
      control: {
        type: "select",
        options: ["md", "lg"],
      },
    },
  },
};

export const Default: Story<ResultProps> = (args) => <Result {...args} />;

Default.args = {
  icon: InfoSVG,
  size: "md",
  title: "This course has no content",
  info: "This course is still under construction and will be ready soon",
};
