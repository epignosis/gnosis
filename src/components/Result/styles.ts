import { css, Theme, SerializedStyles } from "@emotion/react";
import { Size } from "./Result";
import { mq } from "@theme/utils/breakpoints";

export const container = (
  { typeScaleSizes, result }: Theme,
  { size }: { size: Size },
): SerializedStyles => {
  return css`
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 1rem 0;

    .body {
      text-align: center;
      margin-top: 2rem;

      h3 {
        line-height: 1;
        color: ${result.titleColor};
        margin-bottom: 1rem;

        ${mq["sm"]} {
          font-size: ${size === "lg" ? typeScaleSizes.xl : typeScaleSizes.md};
        }
      }

      p {
        color: ${result.infoColor};
      }
    }

    footer {
      margin-top: 1rem;
    }
  `;
};
