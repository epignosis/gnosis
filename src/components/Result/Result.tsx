import React, { FC, ReactNode } from "react";
import { useResponsive } from "@umijs/hooks";
import { SerializedStyles } from "@emotion/react";
import Heading from "../Heading/Heading";
import Text from "../Text/Text";
import { container } from "./styles";
import { IconType } from "types/common";
import { TypographyLevels } from "@theme/utils/typography";

export type Size = "md" | "lg";

export type ResultProps = React.HTMLAttributes<HTMLElement> & {
  icon?: IconType;
  title: string;
  info?: string;
  size?: Size;
  footer?: ReactNode;
};

type IconSize = {
  md: number;
  lg: number;
};

const getIconSize = (isMd: boolean): IconSize => ({
  md: isMd ? 100 : 75,
  lg: isMd ? 180 : 135,
});

const Result: FC<ResultProps> = ({ icon, title, info, footer, size = "lg", ...rest }) => {
  const { md } = useResponsive();
  const Icon = icon;
  const iconSize = getIconSize(md);

  return (
    <article
      data-testid="empty-state-container"
      css={(theme): SerializedStyles => container(theme, { size })}
      {...rest}
    >
      {Icon && <Icon height={iconSize[size]} data-testid="result-icon" />}
      <div className="body">
        <Heading as="h3">{title}</Heading>
        {info && (
          <Text fontSize={size as TypographyLevels} as="p">
            {info}
          </Text>
        )}
      </div>
      {footer && <footer>{footer}</footer>}
    </article>
  );
};

export default Result;
