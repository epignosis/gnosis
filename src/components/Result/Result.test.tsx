import React from "react";
import faker from "faker";
import Result from "./Result";
import { screen, render } from "@test-utils/render";
import { InfoSVG } from "@icons/core";

describe("<Result/>", () => {
  it("renders correctly", () => {
    const { titleTxt, infoTxt, footerTxt } = {
      titleTxt: faker.lorem.sentence(),
      infoTxt: faker.lorem.paragraph(),
      footerTxt: faker.lorem.sentence(),
    };
    render(
      <Result
        title={titleTxt}
        icon={InfoSVG}
        info={infoTxt}
        footer={<button>{footerTxt}</button>}
      />,
    );

    const title = screen.getByText(titleTxt);
    const icon = screen.getByTestId("result-icon");
    const info = screen.getByText(infoTxt);
    const footerBtn = screen.getByText(footerTxt);

    expect(title).toHaveTextContent(titleTxt);
    expect(icon).toBeInTheDocument();
    expect(info).toHaveTextContent(infoTxt);
    expect(footerBtn).toHaveTextContent(footerTxt);
  });

  it("matches snapshot", () => {
    const { container } = render(
      <Result
        title="My title"
        icon={InfoSVG}
        info="My info."
        footer={<button>Action button</button>}
        size="md"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
