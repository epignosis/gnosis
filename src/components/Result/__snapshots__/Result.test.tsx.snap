// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result/> matches snapshot with all props provided 1`] = `
<div>
  <article
    class="css-zb7fnf-container-container"
    data-testid="empty-state-container"
  >
    <svg-file-stub
      data-testid="result-icon"
      height="180"
    />
    <div
      class="body"
    >
      <h3
        class="css-6t4vgf-headingContainer-headingContainer"
      >
        Test title
      </h3>
      <p
        class="css-1gg3twx-textContainer-textContainer"
        data-testid="text-component"
      >
        Test info
      </p>
    </div>
    <footer>
      <button>
        Test footer
      </button>
    </footer>
  </article>
</div>
`;

exports[`<Result/> matches snapshot with footer 1`] = `
<div>
  <article
    class="css-zb7fnf-container-container"
    data-testid="empty-state-container"
  >
    <div
      class="body"
    >
      <h3
        class="css-6t4vgf-headingContainer-headingContainer"
      >
        Test title
      </h3>
    </div>
    <footer>
      <button>
        Test footer
      </button>
    </footer>
  </article>
</div>
`;

exports[`<Result/> matches snapshot with icon 1`] = `
<div>
  <article
    class="css-zb7fnf-container-container"
    data-testid="empty-state-container"
  >
    <svg-file-stub
      data-testid="result-icon"
      height="180"
    />
    <div
      class="body"
    >
      <h3
        class="css-6t4vgf-headingContainer-headingContainer"
      >
        Test title
      </h3>
    </div>
  </article>
</div>
`;

exports[`<Result/> matches snapshot with info 1`] = `
<div>
  <article
    class="css-zb7fnf-container-container"
    data-testid="empty-state-container"
  >
    <div
      class="body"
    >
      <h3
        class="css-6t4vgf-headingContainer-headingContainer"
      >
        Test title
      </h3>
      <p
        class="css-1gg3twx-textContainer-textContainer"
        data-testid="text-component"
      >
        Test info
      </p>
    </div>
  </article>
</div>
`;

exports[`<Result/> matches snapshot with title 1`] = `
<div>
  <article
    class="css-zb7fnf-container-container"
    data-testid="empty-state-container"
  >
    <div
      class="body"
    >
      <h3
        class="css-6t4vgf-headingContainer-headingContainer"
      >
        Test title
      </h3>
    </div>
  </article>
</div>
`;
